Pacemaker/pcs_cluster_36_nodes_feasibility/scale-cluster-upto-36-nodes.md

```diff
@@ -0,0 +1,155 @@
+```
+
+- interleave
+  - It is clone property used in cloning useful for order.
+  - If we want to limit order of resources to same node use this.
+```
+pcs resource create hax clone interleave=true
+```
+
+- Scaling/Adding resources
+  - Best way is to create resource with minimum node. And then scale nodes.
+  - when we create resource with 36 node then network traffics/cpu usage increases quickly. To avoid this problem use below step.
+  - **How to create/start resources in cluster:**
+  ```
+  # Create cib file
+  $ pcs cluster cib cortx-pcs.xml
+
+  # Put one node in standby
+  $ pcs -f cortx-pcs.xml cluster standby srvnode-1
+
+  # Add resources
+  $ pcs -f cortx-pcs.xml resource create myres ocf:heartbeat:Dummy
+
+  # Add nodes but not start them
+  $ pcs -f cortx-pcs.xml cluster node add srvnode-2 --enable
+  $ pcs -f cortx-pcs.xml cluster node add srvnode-3 --enable
+
+  # Now unstandby and start nodes
+  $ pcs -f cortx-pcs.xml cluster unstandby srvnode-1
+  $ pcs -f cortx-pcs.xml cluster start srvnode-2
+  $ pcs -f cortx-pcs.xml cluster start srvnode-3
+
+  # Verify and push file
+  $ pcs cluster verify -V cortx-pcs.xml
+  $ pcs cluster cib-push cortx-pcs.xml
+  ```
+
+- System Utilization (ram/cpu/network)
+  - DC node will use high utilization as it will act as leader and responsible for action.
+  - Utilization depend on no. node and resource going to affect.
+  - Standby 3 node will have less utilization but standby all node will have high utilization.
+
+- s3 and motr like service.
+  - use generic resource to create multiple clone on same node.
+  - we can modify clone size while scaling cluster.
+    ```
+    # Example
+    $ pcs resource create motr ocf:seagate:s3server service=motr unique_clone=true clone clone-max=10 clone-node-max=2 globally-unique=true
+
+    # After scaling change clone size
+    $ pcs resource update motr-clone meta clone-max=20 clone-node-max=2
+    ```
+
+Outcome:
+
+## 1. Add nodes up to 36 verify corosync limit
+- Able to create 36 nodes cluster with 32 normal nodes and 4 remote nodes.
+- We created scripts to create 36 nodes cluster and already shared in the scripts dir.
+
+## 2. Configure the dummy resources with the dependency
+- Configured around 618 resources with the dependecy in the cluster and observed that pcs cluster became unstable.
+- We observed from corosync logs, PCMK_ipc_buffer has exceeded buffer limit and also saw some bad file descriptor related errors.
+- We tunned some parameters to make cluster stable. Please find details in tunning section.
+
+Check PCS Output after configuring the resources (`check pcs_status.txt`)
+
+## 3. Cluster tunning details
+
+Below issue observed after scaling cluster and configuring resources
+- PCMK_ipc_buffer exceeded
+- Bad file descriptor related errors
+
+**Tunning performed**
+1. IPC buffer limit
+- Increased IPC buffer limit (PCMK_ipc_buffer=19725604)
+- IPC buffer is the size of message used to corosync for heartbeat communication.
```

Comment:
used *by corosync?

Reply:
Done. Actually, We used the recommended buffer size in the corosync error logs and after changing the buffer size we are able to get rid of those errors.